app/src/regular/java/com/tari/android/wallet/ui/fragment/settings/backup/WalletBackupSettingsFragment.kt

```diff
@@ -32,48 +32,88 @@
  */
 package com.tari.android.wallet.ui.fragment.settings.backup
 
+import android.animation.Animator
+import android.animation.ValueAnimator
 import android.app.Activity
 import android.content.Context
 import android.content.Intent
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.view.animation.LinearInterpolator
 import androidx.activity.OnBackPressedCallback
 import androidx.appcompat.app.AlertDialog
 import androidx.biometric.BiometricPrompt.ERROR_CANCELED
 import androidx.biometric.BiometricPrompt.ERROR_USER_CANCELED
+import androidx.core.animation.addListener
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.lifecycleScope
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.Scope
 import com.google.api.client.googleapis.extensions.android.gms.auth.UserRecoverableAuthIOException
+import com.google.api.services.drive.DriveScopes
 import com.orhanobut.logger.Logger
+import com.tari.android.wallet.R
+import com.tari.android.wallet.R.color.back_up_settings_permission_processing
 import com.tari.android.wallet.R.string.*
 import com.tari.android.wallet.databinding.FragmentWalletBackupSettingsBinding
+import com.tari.android.wallet.infrastructure.backup.WalletBackup
+import com.tari.android.wallet.infrastructure.backup.storage.BackupStorage
+import com.tari.android.wallet.infrastructure.backup.storage.BackupStorageFactory
 import com.tari.android.wallet.infrastructure.security.biometric.BiometricAuthenticationService
 import com.tari.android.wallet.ui.activity.settings.SettingsRouter
 import com.tari.android.wallet.ui.dialog.ErrorDialog
 import com.tari.android.wallet.ui.extension.*
+import com.tari.android.wallet.ui.util.UiUtil.setColor
+import com.tari.android.wallet.util.SharedPrefsWrapper
 import kotlinx.coroutines.launch
+import org.joda.time.format.DateTimeFormat
+import java.net.UnknownHostException
 import javax.inject.Inject
+import kotlin.coroutines.suspendCoroutine
 
 class WalletBackupSettingsFragment @Deprecated(
     """Use newInstance() and supply all the 
 necessary data via arguments instead, as fragment's default no-op constructor is used by the 
 framework for UI tree rebuild on configuration changes"""
 ) constructor() : Fragment() {
 
+    @Inject
+    lateinit var factory: BackupStorageFactory
+
+    @Inject
+    lateinit var backup: WalletBackup
+
     @Inject
     lateinit var service: BiometricAuthenticationService
 
+    @Inject
+    lateinit var registry: SharedPrefsWrapper
+
+    private val vmObserver = Observer(::handleBackupState)
```

Comment:
Can't understand what `vm` stands for. (Later I found out it's `viewModel`, `viewModelObserver` is good I think.)